### PR TITLE
Add Windows Implementation Libraries as an external library

### PR DESCRIPTION
**Pull request by CookiePLMonster**
Previously a part of #8316 but I would really like to use WIL in other PR's to get RAII wrappers around WinAPI resources.

Quoting the relevant part of this PR:

> As a preparation, [WIL](https://github.com/microsoft/wil) has been added, since it's very useful for ensuring proper RAII-based destruction for several WinAPI types - in my case it was proper scoped calls to `CoUninitialize` and proper destruction of previously leaking (in some cases) `PROPVARIANT`.

WIL also includes proper `unique_ptr`-like wrappers around eg. module handles, file handles, registry handles, `CoTaskMem` memory... etc.

### CODE REVIEW

File: Source/Core/AudioCommon/CMakeLists.txt [hunk 1]
```diff
@@ -71,6 +71,11 @@ if(WIN32)
 
     WASAPIStream.cpp
     WASAPIStream.h
+  target_include_directories(audiocommon PRIVATE
```
Comment (Warepire) [inline Source/Core/AudioCommon/CMakeLists.txt]: Maybe this part should remain with audiocommon or be moved somewhere more generic?

Comment (CookiePLMonster) [reply Source/Core/AudioCommon/CMakeLists.txt]: I don't know if we have a "more generic" place for this, so I just removed it from that PR and will make it part of WASAPI PR.

**Comment by CookiePLMonster** [comment]
As per the discussion on IRC I'll modify cmake to make WIL a dependency library so it's easier to use than what it is now. Will do it around Saturday.

**Comment by CookiePLMonster** [comment]
Yesterday I spent way more time than I'd like trying to fix cmake project on Windows and looks like it's just fundamentally broken - PCH seems to be set up incorrectly and I failed to fix it fully.

Because of this I am unable to test and come up with a reasonable way to integrate WIL there. Any ideas how to proceed?

Tagging @lioncash since they're usually involved in cmake discussions.

**Comment by CookiePLMonster** [comment]
As per discussion on IRC earlier, I now added WIL as a regular directory and not a submodule.

I did notice one problem with that however - now when it's **not** a submodule, it's not obvious what exact revision it contains...

**Comment by shuffle2** [comment]
> I did notice one problem with that however - now when it's **not** a submodule, it's not obvious what exact revision it contains...
In the past dolphin would work around this by leaving a .txt file a level above, stating the version of the external resource, if changes were made, etc. Mainly to simplify process of updating the resource in the future.

**Comment by CookiePLMonster** [comment]
Rebased on latest master.

**Pull request merged**